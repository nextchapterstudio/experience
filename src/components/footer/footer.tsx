import React, { ElementType, FC, HTMLAttributes } from 'react'
import classnames from 'classnames'
import {
  allTheClasses,
  ResponsiveProps,
  isSizedProps,
  BackgroundColorProps,
  ColorProps,
  HasTextProps,
  StyleOrStateProps,
  ScreenProps,
  SizeProps,
  FontWeightProps,
  TextHelpersProps,
} from '../../utils/modifiers'

export interface FooterProps extends HTMLAttributes<HTMLOrSVGElement> {
  children?: React.ReactNode
  className?: string
}
const Footer: FC<
  FooterProps &
    ResponsiveProps &
    isSizedProps &
    BackgroundColorProps &
    ColorProps &
    HasTextProps &
    StyleOrStateProps &
    ScreenProps &
    SizeProps &
    FontWeightProps &
    TextHelpersProps
> = ({ children, className, ...props }) => {
  const classes = allTheClasses(props)
  return <footer className={classnames('footer', className, classes)}>{children}</footer>
}

export { Footer }

export default Footer
