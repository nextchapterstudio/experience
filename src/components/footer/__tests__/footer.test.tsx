import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { Footer } from '../footer'

describe('Container component', () => {
  afterEach(cleanup)
  it('should render', () => {
    const { container } = render(<Footer>Some Text</Footer>)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('footer')
  })
  it('should render with extra classnames', () => {
    const { container } = render(<Footer className="contained">Contained Text</Footer>)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('footer')
    expect(container.firstChild).toHaveClass('contained')
  })
  it('should render with isActive', () => {
    const { container } = render(<Footer isActive>Active Control</Footer>)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('footer')
    expect(container.firstChild).toHaveClass('is-active')
  })
  it('should render with isDanger', () => {
    const { container } = render(<Footer isDanger>Danger Control</Footer>)

    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('footer')
    expect(container.firstChild).toHaveClass('is-danger')
  })
})
