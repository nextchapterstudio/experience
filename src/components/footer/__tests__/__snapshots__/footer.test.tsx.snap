// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component should render 1`] = `
<footer
  class="footer"
>
  Some Text
</footer>
`;

exports[`Container component should render with extra classnames 1`] = `
<footer
  class="footer contained"
>
  Contained Text
</footer>
`;

exports[`Container component should render with isActive 1`] = `
<footer
  class="footer is-active"
>
  Active Control
</footer>
`;

exports[`Container component should render with isDanger 1`] = `
<footer
  class="footer is-danger"
>
  Danger Control
</footer>
`;
